python/processors/nft_marketplace_v2/marketplace_v2_parser.py

```diff
@@ -0,0 +1,270 @@
+            if move_type_address != MARKETPLACE_V2_ADDRESS:
+                continue
+
+            move_type_short = write_set_change_utils.get_move_type_short(
+                write_resource.type
+            )
+            data = json.loads(write_resource.data)
+
+            match move_type_short:
+                case "coin_listing::FixedPriceListing":
+                    price = data.get("price", None)
+                case "listing::Listing":
+                    listing_address = standardize_address(write_resource.address)
+                    index = wsc_index
+                    seller = data.get("seller", None)
+                    seller = standardize_address(seller) if seller else None
+                    inner = data.get("object", {}).get("inner", None)
+
+    # Parse token v1 container or v2 token
+    for _, wsc in enumerate(write_set_changes):
+        write_resource = write_set_change_utils.get_write_resource(wsc)
+        if write_resource:
+            resource_address = write_resource.address
+            if resource_address != inner:
+                continue
+            data = json.loads(write_resource.data)
+            move_type = write_resource.type_str
+
+            if move_type == f"{MARKETPLACE_V2_ADDRESS}::listing::TokenV1Container":
+                token_standard = TokenStandard.TOKEN_V1
+
+                token_data_id_struct = (
+                    data.get("token", {}).get("id", {}).get("token_data_id", {})
+                )
+                collection = token_data_id_struct.get("collection", None)
+                creator = token_data_id_struct.get("creator", None)
+                name = token_data_id_struct.get("name", None)
+
+                token_data_id_type = TokenDataIdType(creator, collection, name)
+                token_data_id = token_data_id_type.to_hash()
+            elif move_type == "0x4::token::Token":
+                token_data_id = resource_address
+
+    if listing_address and token_data_id:
+        token_metadata = token_metadata_mapping[token_data_id]
+
+        current_listing = CurrentNFTMarketplaceListing(
+            token_data_id=token_data_id,
+            listing_address=listing_address,
+            token_name=token_metadata.token_name,
+            collection_id=token_metadata.collection_id,
+            price=price,
+            token_amount=1,
+            token_standard=token_standard.value,
+            seller=seller,
+            is_deleted=False,
+            marketplace=MarketplaceName.EXAMPLE_V2_MARKETPLACE.value,
+            contract_address=transaction_metadata.contract_address,
+            entry_function_id_str=transaction_metadata.entry_function_id_str_short,
+            last_transaction_version=transaction.version,
+            last_transaction_timestamp=transaction_metadata.transaction_timestamp,
+        )
+        listing = NFTMarketplaceListing(
+            transaction_version=transaction.version,
+            index=index,
```

Comment:
you can't mix event_index with wsc_index. What if there are 2 entries and they're the same?

Reply:
Yeah if using event_index this should be negative